### PR TITLE
chore: Allow count of elements in repeated-element array syntax to be any static Field 

### LINKED ISSUE #445: Allow count of elements in repeated-element array syntax to be any `comptime Field`
# Problem

PR #440 expands the repeated-element array syntax to accept a count of elements where `count` is an integer expression containing only literals, basic numeric operations, or global constants. Although this should cover most usecases, it is theoretically overly limiting to users.

# Solution

We could allow all expressions that evaluate to a `comptime Field` (or other countable integer type) as part of the count expression. Then both expressions in `[expr; expr]` would be arbitrary, just with type checking constraints limiting the second to be known at compile-time.

# Alternatives considered

Alternatively, we could say users do not need this flexibility and choose not to have this feature.

### PR DESCRIPTION
**Pull request by f01dab1e**
# Description

WIP

## Problem

Resolves #445

## Summary

WIP

# PR Checklist

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by f01dab1e** [comment]
@jfecher, сan you check this and give me a hint?

File: crates/noirc_evaluator/src/errors.rs [hunk 1]
```diff
@@ -94,7 +94,8 @@ impl From<RuntimeError> for FileDiagnostic {
 impl RuntimeError {
     fn into_diagnostic(self) -> Diagnostic {
         match self {
-            RuntimeError::InternalError(_) => {
```
Comment (f01dab1e) [inline crates/noirc_evaluator/src/errors.rs:97]: maybe you should give more detailed information about the error here?

Comment (jfecher) [reply crates/noirc_evaluator/src/errors.rs:97]: Probably. Looks like there's an empty secondary message we're not using.

**Comment by f01dab1e** [comment]
@jfecher, closed by mistake, can you open it?

**Comment by f01dab1e** [comment]
@jfecher 

For simple literals like `let arr = [1, 2];`, should we indicate the size during desugaring to address type inference issues?
As an analogy in Rust, it's akin to using the turbofish for explicit type annotations:

```rust
slice.as_array::<2>();
```

**Comment by f01dab1e** [comment]
This was blocked by the fact that we had no way to indicate the type (size cannot be inferred from the context), now it's blocked by the same thing, but there's an open pull request after which we can resume work, but we encounter another problem, that the method with desugaring will break the formatter.

**Pull request closed without merging**